src/earn/EarnCollectScreen.tsx

```diff
@@ -0,0 +1,250 @@
+  const poolTokenInfo = useTokenInfo(networkConfig.aaveArbUsdcTokenId)
+  const rewardTokenInfo = useTokenInfo(`${NetworkId['arbitrum-sepolia']}:native`)
+  const feeCurrency = useTokenInfo(`${NetworkId['arbitrum-sepolia']}:native`)
+  if (!tokenInfo || !rewardTokenInfo || !feeCurrency || !poolTokenInfo) {
+    return null
+  }
+
+  const dummyRewardsInfo = [{ amount: new BigNumber(0.00003), tokenInfo: rewardTokenInfo }]
+  const poolApy = 3.47
+  const maxFeeAmount = new BigNumber(0.0001)
+  const onPress = () => {
+    // Todo handle prepared transactions
+  }
+
+  // TODO: Add loading state
+  return (
+    <EarnCollectComponent
+      tokenInfo={tokenInfo}
+      poolTokenInfo={poolTokenInfo}
+      rewardsInfo={dummyRewardsInfo}
+      poolApy={poolApy}
+      maxFeeAmount={maxFeeAmount}
+      feeCurrency={feeCurrency}
+      onPress={onPress}
+    />
+  )
+}
+
+function EarnCollectComponent({
+  tokenInfo,
+  poolTokenInfo,
+  poolApy,
+  rewardsInfo,
+  maxFeeAmount,
+  feeCurrency,
+  onPress,
+}: {
+  tokenInfo: TokenBalance
+  poolTokenInfo: TokenBalance
+  poolApy: number
+  maxFeeAmount: BigNumber
+  feeCurrency: TokenBalance
+  onPress: () => void
+  rewardsInfo: { amount: BigNumber; tokenInfo: TokenBalance }[]
+}) {
+  const { t } = useTranslation()
+  return (
+    <SafeAreaView style={styles.container}>
+      <ScrollView contentContainerStyle={styles.contentContainer}>
+        <Text style={styles.title}>{t('earnFlow.collect.title')}</Text>
+        <View style={styles.collectInfoContainer}>
+          <CollectItem
+            title={t('earnFlow.collect.total')}
+            tokenInfo={tokenInfo}
+            rewardAmount={poolTokenInfo.balance}
+          />
+          {rewardsInfo.map((info, index) => (
+            <CollectItem
+              title={t('earnFlow.collect.total')}
+              key={index}
+              tokenInfo={info.tokenInfo}
+              rewardAmount={info.amount}
+            />
+          ))}
+          <View style={styles.separator} />
+          <Rate tokenInfo={poolTokenInfo} poolApy={poolApy} />
+          <GasFee maxFeeAmount={maxFeeAmount} feeCurrency={feeCurrency} />
+        </View>
+      </ScrollView>
+      <Button
+        style={styles.button}
+        size={BtnSizes.FULL}
+        text={t('earnFlow.collect.cta')}
+        onPress={onPress}
+        testID="EarnCollectScreen/CTA"
+      />
+    </SafeAreaView>
+  )
+}
+
+function CollectItem({
+  tokenInfo,
+  rewardAmount,
+  title,
+}: {
+  tokenInfo: TokenBalance
+  rewardAmount: BigNumber
+  title: string
+}) {
+  return (
+    <>
+      <Text style={styles.collectItemTitle}>{title}</Text>
+      <View style={styles.row}>
+        <View style={styles.iconContainer}>
+          <TokenIcon token={tokenInfo} size={IconSize.LARGE} />
+        </View>
+        <View>
+          <TokenDisplay
+            style={styles.cryptoText}
+            tokenId={tokenInfo.tokenId}
+            amount={rewardAmount || tokenInfo.balance}
```

Comment:
Do we need this fallback? What's a case where it can be falsy?

Reply:
from a previous iteration